src/k_smallest.rs

```diff
@@ -1,20 +1,92 @@
-use alloc::collections::BinaryHeap;
-use core::cmp::Ord;
+use alloc::vec::{Vec, IntoIter};
+use core::cmp::{Ordering, Reverse};
 
-pub(crate) fn k_smallest<T: Ord, I: Iterator<Item = T>>(mut iter: I, k: usize) -> BinaryHeap<T> {
-    if k == 0 { return BinaryHeap::new(); }
+/// Consumes a given iterator, returning the minimum elements in **ascending** order.
+pub(crate) fn k_smallest_general<T, I: Iterator<Item = T>>(
```

Comment:
Probably not sensitive but I think `T` should be replaced by `I::Item` (and just `I: Iterator`).